src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -172,49 +265,262 @@ def selectedshapechange(self):
+            else:
+                self.datafile = QtWidgets.QFileDialog.getOpenFileName(
+                    self, "Choose a file", "","All supported files (*.OMF *.omf *.SLD *.sld);;"
+                                            "OMF files (*.OMF *.omf);;"
+                                            "SLD files (*.SLD *.sld);;"
+                                            "All files (*.*)")[0]
+            # If a file has been sucessfully chosen
             if self.datafile:
+                #set basic data about the file
                 self.default_shape = str(self.cbShape.currentText())
                 self.file_name = os.path.basename(str(self.datafile))
                 self.ext = os.path.splitext(str(self.datafile))[1]
-                if self.ext in self.omf_reader.ext:
+                #select the required loader for the data format
+                if self.ext in self.omf_reader.ext and (not load_nuc):
+                    #only load omf files for magnetic data
                     loader = self.omf_reader
                 elif self.ext in self.sld_reader.ext:
                     loader = self.sld_reader
-                elif self.ext in self.pdb_reader.ext:
+                elif self.ext in self.pdb_reader.ext and load_nuc:
+                    #only load pdb files for nuclear data
                     loader = self.pdb_reader
                 else:
                     loader = None
-                # disable some entries depending on type of loaded file
-                # (according to documentation)
-                if self.ext.lower() in ['.sld', '.pdb']:
-                    self.txtUpFracIn.setEnabled(False)
-                    self.txtUpFracOut.setEnabled(False)
-                    self.txtUpTheta.setEnabled(False)
-                    self.txtUpPhi.setEnabled(False)
 
                 if self.reader is not None and self.reader.isrunning():
                     self.reader.stop()
-                self.cmdLoad.setEnabled(False)
-                self.cmdLoad.setText('Loading...')
+                self.cmdNucLoad.setEnabled(False)
+                self.cmdNucLoad.setText('Loading...')
+                self.cmdMagLoad.setEnabled(False)
+                self.cmdMagLoad.setText('Loading...')
                 self.cmdCompute.setEnabled(False)
                 self.cmdCompute.setText('Loading...')
                 self.communicator.statusBarUpdateSignal.emit(
                     "Loading File {}".format(os.path.basename(
                         str(self.datafile))))
                 self.reader = GenReader(path=str(self.datafile), loader=loader,
-                                        completefn=self.complete_loading_ex,
+                                        completefn=lambda data=None: self.complete_loading_ex(data=data, load_nuc=load_nuc),
                                         updatefn=self.load_update)
                 self.reader.queue()
         except (RuntimeError, IOError):
```

Comment:
I think this is a problem when you try to load any unrecognised file format. I've altered the method in 5ee23cd736af3bc5dedb2ca6a2f494bf6faadd8e so that instead of setting the loader to None and throwing an error it logs an error that the file extension is unsuitable on the log panel, and cancels the load. I think that is probably the best response for the user?